src/dune_engine/build_system.ml

```diff
@@ -471,10 +477,12 @@ end = struct
           in
           let produced_targets =
             match sandbox with
-            | None ->
+            | None -> (
               (* Directory targets are not allowed for non-sandboxed actions, so
                  the call below should not raise. *)
-              Targets.Produced.of_validated_files_exn targets
+              match Targets.Produced.of_validated targets with
```

Comment:
Is the comment above not true any more?

Reply:
Yes, because actions that create directory targets and aren't useful to sandbox can create directory targets with the change above.